src/Message/Analytics/UserLogOnOffSessionMessage.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Message\Analytics;
+
+use App\Message\Analytics\Helper\AnalyticsDataType;
+use DateTimeImmutable;
+use JsonSerializable;
+use Symfony\Component\Uid\Uuid;
+
+class UserLogOnOffSessionMessage extends AnalyticsMessageBase implements JsonSerializable
+{
+
+    public readonly int $userId;
+    public readonly string $userName;
+    public readonly int $gameSessionId;
+    public readonly int $countryId;
+
+    public function __construct(
+        bool $logOn,
+        DateTimeImmutable $timeStamp,
+        Uuid $serverManagerId,
+        int $userId,
+        string $userName,
+        int               $gameSessionId,
+        int               $countryId,
+    ) {
+        parent::__construct(
+            new AnalyticsDataType(
+                $logOn ?
+                    AnalyticsDataType::USER_LOGON_SESSION :
+                    AnalyticsDataType::USER_LOGOFF_SESSION
+            ),
+            $timeStamp,
+            $serverManagerId
+        );
+        $this->userId = $userId;
+        $this->userName = $userName;
+        $this->gameSessionId = $gameSessionId;
+        $this->countryId = $countryId;
+    }
+
+    // phpcs:ignore PSR1.Methods.CamelCapsMethodName.NotCamelCaps
+    public function JsonSerialize() : array
```

Comment:
can we format it for PSR1 without the ignore?

```suggestion
    public function jsonSerialize() : array
```

Reply:
Yes, I tried to do this for the other jsonSerialize functions but I think I missed this one.